x/auth/tx/service.go

```diff
@@ -271,6 +271,22 @@ func (s txServer) TxEncode(ctx context.Context, req *txtypes.TxEncodeRequest) (*
 	}, nil
 }
 
+// TxEncodeAmino implements the ServiceServer.TxEncodeAmino RPC method.
+func (s txServer) TxEncodeAmino(ctx context.Context, req *txtypes.TxEncodeAminoRequest) (*txtypes.TxEncodeAminoResponse, error) {
+	if req.AminoJson == "" {
+		return nil, status.Error(codes.InvalidArgument, "invalid empty tx json")
+	}
+
+	encodedBytes, err := s.clientCtx.LegacyAmino.Marshal(req.AminoJson)
```

Comment:
This is incorrect. We should UnmarshalJSON the req.AminoJSON into a StdTx, then use TxEncoder to marshal the StdTx into bytes